virttest/qemu_devices/qdevices.py

```diff
@@ -1631,14 +1632,25 @@ def _convert_memobj_args(args):
         :return: Converted args.
         :rtype: dict
         """
+        command_dict = {}
         for key, val in args.items():
             if "size" == key:
-                args[key] = int(utils_numeric.normalize_data_size(val, "B"))
+                command_dict[key] = int(utils_numeric.normalize_data_size(val,
+                                                                          "B"))
             # "share" from object( "qom-type": "memory-backend-memfd" )
             # "reserve" from object( "qom-type": "memory-backend-memfd" )
             # "hugetlb" from object( "qom-type": "memory-backend-memfd" )
             elif key in ("share", "reserve", "hugetlb"):
-                args[key] = val == "yes"
+                command_dict[key] = val == "yes"
```

Comment:
hi @nickzhq , I am afraid for share/reserver/hugetlb, we also need to do the same check as below(val.lower() in (yes, on)), pls refer to the following cmd line

-object memory-backend-memfd,id=id,merge=on|off,dump=on|off,share=on|off,prealloc=on|off,size=size,host-nodes=host-nodes,policy=default|preferred|bind|interleave,seal=on|off,hugetlb=on|off,hugetlbsize=size

which comes from man page of qemu-kvm, as we can see on/off are the preferred values, of course, in cfg, we can set the value to yes/no/on/off, then we do the first conversion in set_param function, here is the 2nd time conversion specified for json format. cc @YongxueHong